module/gui/LinkDock.py

```diff
@@ -19,38 +19,210 @@
 
 class NewLinkWindow(QWidget):
     def __init__(self, dock):
         QWidget.__init__(self)
+        self.log = logging.getLogger("guilog")
         self.dock = dock
+        
         self.setLayout(QVBoxLayout())
         layout = self.layout()
         
-        explanationLabel = QLabel("Select a package and then click Add button.")
-        boxLabel = QLabel("Paste URLs here:")
-        self.box = QTextEdit()
+        boxLabel = QLabel(_("Drop or Paste URLs here") + ":")
+        self.box = PlainTextEdit()
         
-        save = QPushButton("Add")
+        self.append = QCheckBox(_("Append URLs"))
+        whatsThis = (self.append.text(), _("Append pasted/dropped text instead of inserting it at the cursor postion."))
+        self.append.setWhatsThis(whatsThisFormat(*whatsThis))
+        whatsThis = (boxLabel.text(), _("One URL (Link) per line.<br>For manual text editing you may want to deactivate") + " '" + self.append.text() + "'.")
+        self.box.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.destQueue     = QRadioButton(_("Queue"))
+        self.destCollector = QRadioButton(_("Collector"))
+        destBtnLayout = QHBoxLayout()
+        destBtnLayout.addWidget(self.destQueue)
+        destBtnLayout.addWidget(self.destCollector)
+        destBtnLayout.addStretch(1)
+        self.destAutoSelect = QCheckBox(_("Select with tab"))
+        
+        self.clear = QPushButton(_("Clear"))
+        whatsThis = (self.clear.text(), _("Clears the URL box."))
+        self.clear.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.filter = QPushButton()
+        
+        self.save = QPushButton(_("Add"))
+        self.save.setIcon(self.dock.appIconSet["add_small"])
+        self.msg = QLabel("ERROR")
+        s = "QWidget { color: crimson; background-color: %s }" % QColor(Qt.gray).name()     # red text color on gray background
+        self.msg.setStyleSheet(s)
+        lsp = self.msg.sizePolicy()
+        lsp.setHorizontalPolicy(QSizePolicy.Ignored)
+        self.msg.setSizePolicy(lsp)
+        hbox = QHBoxLayout()
+        hbox.addWidget(self.msg)
+        hbox.addWidget(self.save)
         
-        layout.addWidget(explanationLabel)
         layout.addWidget(boxLabel)
         layout.addWidget(self.box)
-        layout.addWidget(save)
+        layout.addWidget(self.append)
+        layout.addWidget(self.filter)
+        layout.addWidget(self.clear)
+        layout.addLayout(destBtnLayout)
+        layout.addWidget(self.destAutoSelect)
+        layout.addLayout(hbox)
+        
+        self.adjustSize()
+        self.msg.setFixedHeight(self.save.height())
+        self.slotMsgHide()
         
-        self.connect(save, SIGNAL("clicked()"), self.dock.slotDone)
+        self.connect(self.save, SIGNAL("clicked()"), self.dock.slotDone)
+        self.connect(self.filter, SIGNAL("clicked()"), self.dock.slotFilterBtnClicked)
+        self.connect(self.box, SIGNAL("textChanged()"), self.dock.slotBoxTextChanged)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['SIGNAL' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)